nntrainer/layers/concat_layer.cpp

```diff
@@ -118,25 +118,50 @@ void ConcatLayer::forwarding(RunLayerContext &context, bool training) {
   output.reshape(output_reshape_helper);
   unsigned int output_height_offset = 0;
   unsigned int data_copy_size = output_reshape_helper.width();
+  TensorDim::TensorType tensor_type = out_dim.getTensorType();
 
   for (unsigned int idx = 0; idx < context.getNumInputs(); idx++) {
     Tensor &input = context.getInput(idx);
     const TensorDim in_dim = input.getDim();
     auto const &irh = input_reshape_helper[idx];
     input.reshape(irh);
 
-    /** loop over the dimensions before the concat dimension */
-    for (unsigned int batch = 0; batch < output.batch(); batch++) {
-      /** loop over the concat dimension itself */
-      for (unsigned int count = 0; count < irh.height(); count++) {
-        Tensor dest_tensor = Tensor::Map<float>(
-          output.getAddress<float>(batch, 0, output_height_offset + count, 0),
-          data_copy_size * sizeof(float), {1, 1, 1, data_copy_size});
-        const Tensor source_tensor = Tensor::Map<float>(
-          input.getAddress(batch, 0, count, 0), data_copy_size * sizeof(float),
-          {1, 1, 1, data_copy_size});
-        dest_tensor.copy(source_tensor);
+    if (in_dim.getDataType() == TensorDim::DataType::FP32) {
```

Comment:
Personally, I’m curious.

This part seems almost similar to the code after line 238 below.
Wouldn’t it be possible to separate it into a function?!

Reply:
You mean using `ConcatLayer::forwarding` inside of `ConcatLayer::calcDerivative` ?
If your point is right, that would reduce code length. However, it is not the main purpose of this PR...
Maybe we can make a separate PR after double-checking..!